docs/architecture/dapr-for-net-developers/service-invocation.md

```diff
@@ -54,113 +53,222 @@ Because the calls flow through sidecars, Dapr can inject some useful cross-cutti
+
+```csharp
+var daprClient = new DaprClientBuilder().Build();
+try
+{
+    var confirmation = await daprClient.InvokeMethodGrpcAsync<Order, OrderConfirmation>("orderservice", "submitOrder", order);
+}
+catch (InvocationException ex)
+{
+    // Handle error
+}
+```
+
+In the example above, DaprClient serializes the given `order` object using [Protobuf](https://developers.google.com/protocol-buffers) and uses the result as the gRPC request body. Likewise, the response body is Protobuf deserialized and returned to the caller. Protobuf typically provides better performance than the JSON payloads used in HTTP service invocation.
+
+## Reference application: eShopOnDapr
+
+The original [eShopOnContainers](https://github.com/dotnet-architecture/eShopOnContainers) microservice reference architecture from Microsoft used a mix of HTTP/REST and gRPC services. The use of gRPC was limited to communication between an [aggregator service](../cloud-native/service-to-service-communication#service-aggregator-pattern) and core backend services. Figure 6-2 show the architecture:
+
+![gRPC and HTTP/REST calls in eShopOnContainers](./media/service-invocation/eshop-on-containers.png)
 
 **Figure 6-2**. gRPC and HTTP/REST calls in eShopOnContainers.
 
 Note the steps from the previous figure:
 
-1. The front-end calls the [API Gateway](https://docs.microsoft.com/azure/architecture/microservices/design/gateway) using HTTP/REST.
+1. The frontend calls the [API gateway](https://docs.microsoft.com/azure/architecture/microservices/design/gateway) using HTTP/REST.
 
-1. The API Gateway forwards simple [CRUD](https://www.sumologic.com/glossary/crud/) (Create, Read, Update, Delete) requests directly to a core back-end service using HTTP/REST.
+1. The API gateway forwards simple [CRUD](https://www.sumologic.com/glossary/crud/) (Create, Read, Update, Delete) requests directly to a core backend service using HTTP/REST.
 
-1. The API Gateway forwards complex requests that involve coordinated calls to multiple back-end services to the Web Shopping Aggregator service.
+1. The API gateway forwards complex requests that involve coordinated calls to multiple backend services to the web shopping aggregator service.
 
-1. The aggregator service uses gRPC to call core back-end services.
+1. The aggregator service uses gRPC to call core backend services.
 
-In the recently updated eShopOnDapr implementation, Dapr sidecars are added to the services and API Gateway. Figure 6-3 show the updated architecture:
+In the recently updated eShopOnDapr implementation, Dapr sidecars are added to the services and API gateway. Figure 6-3 show the updated architecture:
 
-![gRPC and HTTP/REST calls in eShopOnContainers](./media/service-invocation/eshopondapr.png)
+![gRPC and HTTP/REST calls in eShopOnContainers](./media/service-invocation/eshop-on-dapr.png)
```

Comment:
```suggestion
![gRPC and HTTP/REST calls with sidecars in eShopOnContainers](./media/service-invocation/eshop-on-dapr.png)
```

Fix warning about duplicate ALT text, we need to take these warnings very seriously. Eventually they'll be hard errors, and we need to be mindful of screen-readers. We want to be descriptive of our imagery.